gen/drupal_elixir

```diff
@@ -175,42 +175,39 @@ sub processUsers {
 	if(@admins) {
 		foreach my $user (@admins){
 			if ($user->getId() == $uid) {
-				processMemberships $gid, $uid, $A_M_TYPE_ADMIN;
+				processMemberships $groupId, $uid, $A_M_TYPE_ADMIN;
 			}
 		}
 	}
 
-	processMemberships $gid, $uid, $A_M_TYPE_MEMBER;
+	processMemberships $groupId, $uid, $A_M_TYPE_MEMBER;
 }
 
 ## creates structure for groups.csv file
 sub processGroups {
-	my $group = shift;
-	my %groupAttributes = attributesToHash $group->getAttributes;
-	my $membersElement = ($group->getChildElements)[1];
-
-	if($groupAttributes{$A_GROUP_NAME}) {
-		my $groupName = $groupAttributes{$A_GROUP_NAME};
-		my $gid = $groupAttributes{$A_GROUP_ID};
-		my $groupParentID = $groupAttributes{$A_PARENT_GROUP_ID};
+	my ($groupId, $resourceId) = @_;
+
+	if($data->getGroupAttributeValue( group => $groupId, attrName => $A_GROUP_NAME )) {
+		my $groupName = $data->getGroupAttributeValue( group => $groupId, attrName => $A_GROUP_NAME ),
+		my $groupParentID = $data->getGroupAttributeValue( group => $groupId, attrName => $A_PARENT_GROUP_ID );
 		my $groupType;
 
-		if($groupAttributes{$A_GROUP_TYPE}) {
-			$groupType = $groupAttributes{$A_GROUP_TYPE};
+		if($data->getGroupResourceAttributeValue( group => $groupId, resource=> $resourceId, attrName => $A_GROUP_TYPE )) {
+			$groupType = $data->getGroupResourceAttributeValue( group => $groupId, attrName => $A_GROUP_TYPE );
```

Comment:
Since attribute is group-resource, you must pass also `resource => $resourceId` param to the method.

Reply:
Fixed